src/main/java/runwar/undertow/WebXMLParser.java

```diff
@@ -172,8 +190,14 @@ public static void parseWebXml(File webxml, File webinf, DeploymentInfo info,
             });
             info.addFilters(filterMap.values());
 
+            // do filter mappings
             Match filterMappings = $(doc).find("filter-mapping");
-            trace("Total No. of filters-mappings: %s", filterMappings.size());
+            int sizeFilterMappings = filterMappings.size();
+            trace("Total No. of filters-mappings: %s", sizeFilterMappings);
+            if ( overrideEnabled && sizeFilterMappings > 0 ) {
+                trace("Removing %s filter mapping(s) from deployment due to web.xml override", info.getFilterMappings().size());
+                info.getFilterMappings().clear();
```

Comment:
@BalbinoOrtus This will not work?  Remember, this is the entire reason for the pull we are sending to Undertow.  To allow this to work.  The List you are clearing here will not mutate the state of the `DeploymentInfo` class as this getting simply returns a new List with a copy of the filter mappings.  This needs removed or commented out until a later date once Undertow has merged our pull and we are using that new version of Undertow and can directly access and clear the two different internal Lists of filter mappings.